Significant-Gravitas/Auto-GPT
# Switch to Pyenv and Poetry for Dependency Management

### Background

To improve our project's dependency management and make it easier for contributors to set up their development environments, we're switching to **pyenv** for Python version management and **poetry** for package management. These tools allow for more consistent and streamlined dependency handling and simplify the process of installing and updating packages.

### Changes

- Updated the project documentation to include instructions for setting up pyenv and poetry
- Removed requirements.txt and added a pyproject.toml file for dependency management with poetry

### Documentation

The project documentation has been updated to include instructions on setting up **pyenv** and **poetry**. These instructions can be found in the **README.md** file and cover the installation and usage of both tools.

### Test Plan
To test these changes, follow these steps:

- Set up a new pyenv virtual environment following the instructions in the README.md file
- Install poetry and use it to install the project dependencies
- Ensure that the project runs as expected with the new dependency management setup
- Verify that the poetry lock file (poetry.lock) is generated and works as expected


### PR Quality Checklist
